qupulse/program/waveforms.py

```diff
@@ -215,7 +220,17 @@ def reversed(self) -> 'Waveform':
         """Returns a reversed version of this waveform."""
         # We don't check for constness here because const waveforms are supposed to override this method
         return ReversedWaveform(self)
-
+    
+    @abstractmethod
+    def _compare_subset_key(self, channel_subset: Set[ChannelID]) -> Hashable:
```

Comment:
What is the reason for this method? The current implementation is worse than `hash(wf.get_subset_for_channels(channel_subset))`. If we need better subset analysis, we can improve waveform normalization.

Reply:
this was done before the compare_key thing was reworked, i believe.
but your code would take into account channel names, right? this is what i tried to avoid there to just compare potential sample output.

Reply:
this change however is also not relevant for the linspace thing and propagated to the branch from somewhere